src/erlcloud_aws.erl

```diff
@@ -1562,3 +1565,10 @@ get_env_for_role_credentials(Arn, ExtId) ->
 set_env_for_role_credentials(Arn, ExtId, Val) ->
     % application:set_env is undocumented in regards to type(Par) =/= atom()
     application:set_env(erlcloud, {role_credentials, Arn, ExtId}, Val).
+
+-spec maybe_imdsv2_session_token(aws_config()) -> binary() | undefined.
+maybe_imdsv2_session_token(Config) ->
+    case erlcloud_ec2_meta:generate_session_token(60, Config) of
```

Comment:
let's fix this in test and good to go.

Failure/Error: {error,undef,
                        [{erlcloud_ec2_meta,generate_session_token,
                             [60,
                              {aws_config,